narwhals/dtypes.py

```diff
@@ -751,16 +774,24 @@ def __eq__(self: Self, other: DType | type[DType]) -> bool:  # type: ignore[over
         if type(other) is type and issubclass(other, self.__class__):
             return True
         elif isinstance(other, self.__class__):
-            return self.inner == other.inner
+            if self.shape != other.shape:
+                return False
+            else:
+                return self.inner == other.inner
         else:
             return False
 
     def __hash__(self: Self) -> int:
-        return hash((self.__class__, self.inner, self.width))
+        return hash((self.__class__, self.inner, self.shape))
+
+    def __repr__(self) -> str:
+        # Get leaf type
+        dtype = self.inner
+        while isinstance(dtype, Array):
+            dtype = dtype.inner
 
-    def __repr__(self: Self) -> str:
         class_name = self.__class__.__name__
-        return f"{class_name}({self.inner!r}, {self.width})"
+        return f"{class_name}({dtype!r}, shape={self.shape})"
```

Comment:
This diff is that's causing marimo ci to fail

Reply:
It also wouldn't be too bad to override repr in v1 right?

Reply:
Should we maintain `width` in v1 then? My understanding is that we are limited to 1d arrays with the v1 implementation.
Or should we just keep the `__repr__` and show `f"{class_name}({self.inner!r}, {width})"` with `width=self.shape[0]` if shape has only one dimension else the tuple `self.shape`?